fixed the character \" escaping problem in conanbuildinfo.xmake.lua generator

uikino opened the pull request:
# bug修复
修复issue https://github.com/xmake-io/xmake/issues/1868#issue-1066712228 中导致的转义字符问题
# 修复方法
Linux 文件系统并未禁止绝大多数特殊字符
所以对于路径 宏定义 应用修改
创建```prepare_process_escape_character```函数处理转义字符
创建类```XmakeDepsFormatter```成员函数```__filter_char```调用```prepare_process_escape_character```
对以下类内成员应用修改
```
self.include_paths
self.lib_paths
self.bin_paths
self.defines
self.rootpath
```

uikino committed:
fixed the character " escaping problem

uikino committed:
revoke modification for flags and lib name

waruqi commented on conanfile.py:
```diff
@@ -57,17 +57,26 @@ def content(self):
+def prepare_process_escape_character(raw_string):
+    if raw_string.find('\"') != -1:
+        raw_string = raw_string.replace("\"","\\\"")
+    return raw_string
```
为啥不直接把实现放到 __filter_char 里面去呢，还额外搞个 全局 函数？

uikino replied:
随便，我这样做只是想那天又要过滤其他字符就优点麻烦

waruqi replied:
可以把它挪到class里作为成员函数，尽量不要放到全局，这跟过滤其他字符不冲突

uikino replied:
已经做出修改了

uikino committed:
move the prepare fun into class

move the function prepare_process_escape_character to XmakeDepsFormatter and rename it to __prepare_process_escape_character

waruqi commented:
已 merge，谢谢
